snap: update for core24

soumyaDghosh opened the pull request.

GeopJr commented on .github/workflows/build.yml:
```diff
@@ -69,15 +68,18 @@ jobs:
       if: matrix.arch == 'aarch64'
-    - uses: diddlesnaps/snapcraft-multiarch-action@v1
-      if: matrix.arch == 'aarch64'
+    - uses: diddledani/snapcraft-multiarch-action@v1
```
The action is still under `diddlesnaps/snapcraft-multiarch-action`

soumyaDghosh replied:
Ooo sorry, I mistakenly changed it.

GeopJr commented on .github/workflows/build.yml:
```diff
@@ -69,15 +68,18 @@ jobs:
       if: matrix.arch == 'x86_64'
-    - uses: actions/upload-artifact@v3
+      env:
+        SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: 1
```
I can't ship it with this, can I?

soumyaDghosh replied:
Yes you can, but for now only amd64 and just from github actions.

GeopJr commented on build-aux/snap/snapcraft.yaml:
```diff
@@ -1,9 +1,9 @@
 adopt-info: tuba
 
-grade: devel
+grade: stable
```
Is this needed?

soumyaDghosh replied:
devel snaps can't be pushed to stable to candidate channels

GeopJr replied:
This is only for devel purposes though, the release config lives at https://github.com/GeopJr/tuba-snap/

soumyaDghosh replied:
We can directly publish this snap to stable from here. As the snap is far behind right now.

soumyaDghosh replied:
Added workflow to publish this to store.

GeopJr commented:
Thanks for this PR!

FWIW, this action and snap are only for ci artifacts and development builds. The release snap config lives at https://github.com/GeopJr/tuba-snap/ (you don't have to open a PR there, I'll pick the changes from here!)

soumyaDghosh commented:
> Thanks for this PR!
> 
> FWIW, this action and snap are only for ci artifacts and development builds. The release snap config lives at https://github.com/GeopJr/tuba-snap/ (you don't have to open a PR there, I'll pick the changes from here!)

You cannot directly build this in the `snapcraft.io` website builder. Because gnome extension still experimental.

GeopJr commented:
:/ disappointing

soumyaDghosh commented:
> :/ disappointing

Actually the extensions are getting a major overhaul on the graphics side, with new graphics drivers. So, it's taking this long.

https://github.com/canonical/snapcraft/pull/4744

soumyaDghosh commented:
You can get the login cradentials via this command,

```bash
snapcraft export-login --snaps=tuba \
  --acls package_access,package_release \
  SNAP_STORE_STABLE
```

Then set that in an environment variable named `STORE_LOGIN`

GeopJr commented:
I'll do so when I get on my snap environment! Thanks!

I have one more question though, if it compiles with -Ddevel=true, how do I make 'stable' builds from here? If I promote the edge one, it will be treated as devel and the action is not aware of repo releases. That's one of the reasons I split it into devel snap config and release snap config in a different repo.

soumyaDghosh commented:
> I'll do so when I get on my snap environment! Thanks!
> 
> I have one more question though, if it compiles with -Ddevel=true, how do I make 'stable' builds from here? If I promote the edge one, it will be treated as devel and the action is not aware of repo releases. That's one of the reasons I split it into devel snap config and release snap config in a different repo.

Okay, so even more conditioning! Ummmm... For now I removed the devel using sed, but I think better you could do is replicate this in the snap manifests repo. And remove things from here which you don't need anymore.

GeopJr commented:
Here's an idea: How about I unlink the tuba-snap repo from the snapcraft runners and we instead add an action there with (workflow_dispatch so I can trigger it manually) to build and publish snaps?

That way, the action on this repo can stay as devel and the action on tuba-snap will always be release only

GeopJr commented:
Merging, thanks!

I don't know if/when I'll set up action publishing on the tuba-snap repo.

GeopJr commented:
Thanks for letting me know!

Can I drop `SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`?

soumyaDghosh commented:
No Sorry, my bad, not yet. It's not yet stable. I hope in the next release of snapcraft, it'll be. And yes, you can drop it once stable
